pkg/generate/app/sourcelookup.go

```diff
@@ -215,18 +234,37 @@ func (e SourceRepositoryEnumerator) Detect(dir string) (*SourceRepositoryInfo, e
 	return info, nil
 }
 
-func StrategyAndSourceForRepository(repo *SourceRepository) (*BuildStrategyRef, *SourceRef, error) {
-	remoteUrl, err := repo.RemoteURL()
+// StrategyAndSourceForRepository returns the build strategy and source code reference
+// of the provided source repository
+func StrategyAndSourceForRepository(repo *SourceRepository, resolver Resolver) (*BuildStrategyRef, *SourceRef, error) {
+	srcRef, err := NewSourceRefGenerator().FromGitURL(repo.location)
 	if err != nil {
-		return nil, nil, fmt.Errorf("cannot obtain remote URL for repository at %s", repo.location)
+		return nil, nil, fmt.Errorf("cannot obtain remote URL for repository at %s: %v", repo.location, err)
 	}
-	strategy := &BuildStrategyRef{
-		IsDockerBuild: repo.IsDockerBuild(),
+	strategy, err := NewBuildStrategyRefGenerator(source.DefaultDetectors, resolver).FromSourceRef(srcRef)
```

Comment:
I don't like doing resolution in these methods.  All resolution should happen before this method gets called (specifically for new-app) at the very early part of the flow.